### PR TITLE
Switch to BlocProvider's

### PR DESCRIPTION
**Pull request by AlexanderSimakov**

### CODE REVIEW

File: lib/src/data/repositories/suggestion_repository.dart [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 import 'package:suggest_a_feature/src/data/interfaces/cache_data_source.dart';
 import 'package:suggest_a_feature/src/data/interfaces/suggestions_data_source.dart';
-import 'package:suggest_a_feature/src/domain/data_interfaces/suggestion_repository.dart';
+import 'package:suggest_a_feature/src/domain/data_interfaces/i_suggestion_repository.dart';
 import 'package:suggest_a_feature/src/domain/entities/comment.dart';
 import 'package:suggest_a_feature/src/domain/entities/suggestion.dart';
 import 'package:suggest_a_feature/src/domain/entities/suggestion_author.dart';
 
-class SuggestionRepositoryImpl implements SuggestionRepository {
+class SuggestionRepositoryImpl implements ISuggestionRepository {
```
Comment (flutterwtf) [inline lib/src/data/repositories/suggestion_repository.dart:9]: We don't need prefix here

File: lib/src/presentation/pages/suggestion/suggestion_cubit.dart [hunk 2]
```diff
@@ -98,6 +108,12 @@ class SuggestionCubit extends Cubit<SuggestionState> {
     }
   }
 
+  @override
+  Future<void> close() async {
+    dispose();
+    await super.close();
+  }
+
   void dispose() {
```
Comment (flutterwtf) [inline lib/src/presentation/pages/suggestion/suggestion_cubit.dart:117]: we don't need this function, I guess. We already have `close()` for such operations

File: lib/src/data/repositories/suggestion_repository.dart (hunk 1 above)
Comment (AlexanderSimakov) [reply lib/src/data/repositories/suggestion_repository.dart]: Fixed!

File: lib/src/presentation/pages/suggestion/suggestion_cubit.dart (hunk 2 above)
Comment (AlexanderSimakov) [reply lib/src/presentation/pages/suggestion/suggestion_cubit.dart]: Fixed!

**Review by flutterwtf — APPROVED** [review]

**Pull request merged**